### DISCUSSION TITLE
REQUEST: Identify orgs accessed by an outside collaborator in "List enterprise consumed licenses"

### DISCUSSION THREAD

**Discussion by ksprinkl** [community/community#77317] [Enterprise]
### Select Topic Area

Product Feedback

### Body

In order to understand how licenses are being consumed by the orgs in an enterprise, we need to know the orgs that an outside collaborator has access to. Ideally the "List enterprise consumed licenses" api would return the orgs for an outside collaborator in the same way it returns the orgs for a member.   Outside collaborators are returned but the org field is empty.  We need to know which orgs are providing access that requires license for the outside collaborator.

**Comment by CheetahCodes21** [comment]
To enhance the "List enterprise consumed licenses" API to include the organizations that an outside collaborator has access to, the following modifications can be implemented:

1. **Expand the API Response:** The API response should be modified to include an additional field, "collaborator_organizations," for each outside collaborator entry. This field would contain a list of organization IDs that the outside collaborator has access to.

2. **Enrich Organization Data:** When an outside collaborator is invited to join an organization, the organization ID should be associated with the collaborator's account. This information can be stored in a separate table or integrated into the existing user profile structure.

3. **Retrieve Organization Data:** When fetching license consumption data for outside collaborators, the API should retrieve the associated organization IDs from the user profile or the dedicated table.

4. **Populate Collaborator Organizations:** For each outside collaborator entry in the API response, the associated organization IDs should be retrieved and added to the "collaborator_organizations" field.

5. **Return Enhanced Response:** The API should return the modified response structure, including the "collaborator_organizations" field for each outside collaborator entry.

By implementing these modifications, the "List enterprise consumed licenses" API would provide a comprehensive view of license consumption, including the organizations that outside collaborators have access to. This information would be valuable for understanding how licenses are being utilized across the enterprise and identifying potential areas for optimization.
